### PR TITLE
Add: Implement create and list endpoint

### PR DESCRIPTION
**Pull request by mgoetzegb**
## What

Implement create and list notifications endpoint. Created notifications are stored in the database and can be retrieved later via list.

Filtering and Sorting is not yet supported.

Note: The packages `errs` and `restErrorHandler` as well as the function `PrepareResultSelector` was duplicated from the VI backend repo. Ultimately this code should be moved to [greenbone/opensight-golang-libraries](https://github.com/greenbone/opensight-golang-libraries).

## Why

The service is supposed to do something.

## References

VTI-151

## Checklist

- [x] Tests

### CODE REVIEW

**Review by endyApina — COMMENTED** [review]
LGTM, just some comments on the license headers

**Comment by endyApina** [comment]
Also, does it make send to extract the `errs` pkg to the `golang-library`
Since it's the same we use in the [VI](https://github.com/greenbone/vulnerability-intelligence-backend/blob/main/pkg/errs/errors.go)

File: pkg/web/cors_middleware.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// Copyright (C) Greenbone Networks GmbH
```
[the remark this answers is not in the document]
Comment (mgoetzegb) [reply pkg/web/cors_middleware.go]: changed to 2024 and to AG

File: pkg/web/helper/parameter_parsing.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
+// SPDX-FileCopyrightText: 2023 Greenbone AG <https://greenbone.net>
```
[the remark this answers is not in the document]
Comment (mgoetzegb) [reply pkg/web/helper/parameter_parsing.go]: changed to 2024

**Pull request merged**